onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -1037,6 +1054,10 @@ Status QnnBackendManager::SetHtpPowerConfig(uint32_t htp_power_config_client_id,
 
 Status QnnBackendManager::SetRpcControlLatency(uint32_t htp_power_config_client_id,
                                                uint32_t rpc_control_latency) {
+  // This function is called in QNN EP's OnRunStart() even if QNN backend setup failed and the model is assigned
+  // to a different EP. Therefore, we have to check that backend setup actually completed before trying to
+  // set RPC control latency. Otherwise, this causes a segfault because the QNN backend library is unloaded.
+  ORT_RETURN_IF_NOT(backend_setup_completed_, "Cannot set HTP RPC control latency if backend setup is not complete.");
```

Comment:
If we are going to use `backend_setup_completed_` for this purpose, should the `backend_setup_completed_ = false` be moved to the beginning of `ReleaseResources()`?

Reply:
These functions (CreateHtpPowerCfgId, SetHtpPowerConfig, and SetRpcControlLatency) are called only on session run. By this time, backend setup has already completed (or failed).

If we ever change it so that the backend setup happens outside of session creation (model load), then we would need to protect all access to these handles with a mutex. If it's alright with you, I'd prefer to handle this separately.